src/encoder/encoderfdkaac.cpp

```diff
@@ -0,0 +1,408 @@
+
+    kLogger.debug() << "Loaded libfdk-aac";
+}
+
+EncoderFdkAac::~EncoderFdkAac() {
+    aacEncClose(&m_aacEnc);
+
+    if(m_pAacDataBuffer)
+        delete m_pAacDataBuffer;
+
+    if (m_library && m_library->isLoaded()) {
+        flush();
+        m_library->unload();
+        delete m_library;
+        kLogger.debug() << "Unloaded libfdk-aac";
+    }
+
+    if (m_pFifoChunkBuffer)
+        delete m_pFifoChunkBuffer;
+
+    if (m_pInputFifo)
+        delete m_pInputFifo;
+}
+
+QString EncoderFdkAac::buttWindowsFdkAac() {
+    // Return %APPDATA%/Local path
+    QString appData = QDesktopServices::storageLocation(
+            QDesktopServices::DataLocation);
+    appData = QFileInfo(appData).absolutePath();
+
+    // Candidate paths for a butt installation
+    QStringList searchPaths;
+    searchPaths << "C:/Program Files";
+    searchPaths << "C:/Program Files (x86)";
+    searchPaths << appData;
+
+    // Try to find a butt installation in one of the
+    // potential paths above
+    for (QString topPath : searchPaths) {
+        QDir folder(topPath);
+        if(!folder.exists()) {
+            continue;
+        }
+
+        // Typical name for a butt installation folder
+        // is "butt-x.x.x" so list subfolders beginning with "butt"
+        QStringList nameFilters("butt*");
+        QStringList subfolders =
+                folder.entryList(nameFilters, QDir::Dirs, QDir::Name);
+
+        // If a butt installation is found, try
+        // to find libfdk-aac in it
+        for (QString subName : subfolders) {
+            if(!folder.cd(subName)) {
+                continue;
+            }
+
+            kLogger.debug()
+                << "Found potential B.U.T.T installation at"
+                << (topPath + "/" + subName);  
+
+            QString libFile = "libfdk-aac-1.dll";
+            if(folder.exists(libFile)) {
+                // Found a libfdk-aac here.
+                // Return the full path of the .dll file.
+                return folder.absoluteFilePath(libFile);
+            }
+
+            folder.cdUp();
+        }
+    }
+
+    return QString::null;
+}
+
+void EncoderFdkAac::setEncoderSettings(const EncoderSettings& settings) {
+    // TODO(Palakis): support more bitrate configurations
+    m_bitrate = settings.getQuality();
+    switch (settings.getChannelMode()) {
+        case EncoderSettings::ChannelMode::MONO:
+            m_channels = 1;
+            break;
+        case EncoderSettings::ChannelMode::STEREO:
+            m_channels = 2;
+            break;
+        case EncoderSettings::ChannelMode::AUTOMATIC:
+            m_channels = 2;
+            break;
+    }
+}
+
+int EncoderFdkAac::initEncoder(int samplerate, QString errorMessage) {
+    (void)errorMessage;
+    m_samplerate = samplerate;
+
+    if(!m_library) {
+        kLogger.warning() << "initEncoder failed: fdk-aac library not loaded";
+        return -1;
+    }
+
+    // This initializes the encoder handle but not the encoder itself.
```

Comment:
Do we need error checking for any of the function calls below? Or is the `aacEncEncode` the only way we can know if these methods failed?